src/types/observablearray.ts

```diff
@@ -119,6 +119,29 @@ function inherit(ctor, proto) {
 }
 inherit(StubArray, Array.prototype)
 
+// Weex freeze Array.prototype
```

Comment:
Thanks for the PR.

Could you make the patch conditional? That is, only apply the patch if the properties are actually not writable (such as in the weex case)? This avoids putting more methods in the prototype chain when it is not really needed.